volmdlr/faces.py

```diff
@@ -5672,258 +5641,381 @@ def is_inside_portion(cutting_contour, inner_contour_spliting_points1, inner_con
+            if over_self_outer_contour and over_planeface_outer_contour:
+                continue
+            if self.edge3d_inside(linesegment3d) or over_self_outer_contour:
+                if planeface.edge3d_inside(linesegment3d):
+                    planeface_intersections.append(volmdlr.wires.Wire3D([linesegment3d]))
+                elif over_planeface_outer_contour:
+                    planeface_intersections.append(volmdlr.wires.Wire3D([linesegment3d]))
+        return planeface_intersections
+
+    def triangle_intersections(self, triangleface):
+        return self.planeface_intersections(triangleface)
+
+    def cylindricalface_intersections(self, cylindricalface: 'CylindricalFace3D'):
+        cylindricalsurfaceface_intersections = cylindricalface.surface3d.plane_intersection(self.surface3d)
+        if not isinstance(cylindricalsurfaceface_intersections[0], vme.Line3D):
+            if all(self.edge3d_inside(intersection) and cylindricalface.edge3d_inside(intersection)
+                   for intersection in cylindricalsurfaceface_intersections):
+                return cylindricalsurfaceface_intersections
+        intersections_points = self.face_intersections_outer_contour(cylindricalface)
+        for point in cylindricalface.face_intersections_outer_contour(self):
+            if point not in intersections_points:
+                intersections_points.append(point)
+        face_intersections = []
+        for primitive in cylindricalsurfaceface_intersections:
+            points_on_primitive = []
+            for point in intersections_points:
+                if primitive.point_belongs(point):
+                    points_on_primitive.append(point)
+            if not points_on_primitive:
+                continue
+            if isinstance(primitive, vme.Line3D):
+                points_on_primitive = primitive.sort_points_along_line(points_on_primitive)
             else:
-                if new_face_contour not in valid_new_faces_contours:
-                    inner_contours = []
-                    for inner_contour in self.surface2d.inner_contours:
-                        if new_face_contour.is_inside(inner_contour):
-                            inner_contours.append(inner_contour)
-                    valid_new_faces_contours.append(new_face_contour)
-                    valid_inner_contours.append(inner_contours)
-        return valid_new_faces_contours, valid_inner_contours
+                points_on_primitive = primitive.sort_points_along_wire(points_on_primitive)
+                points_on_primitive = points_on_primitive + [points_on_primitive[0]]
+            for point1, point2 in zip(points_on_primitive[:-1], points_on_primitive[1:]):
+                edge = primitive.trim(point1, point2)
+                if self.edge3d_inside(edge) and cylindricalface.edge3d_inside(edge):
+                    face_intersections.append(volmdlr.wires.Wire3D([edge]))
+        return face_intersections
```

Comment:
method to verify the intersections between a planeface3D and a cylindrical face3D